File: API/routes/paciente.js
const express = require('express');
const router = express.Router();
const db = require('../db');

// ------------------- LISTAR / BUSCAR PACIENTES ------------------- //
router.get('/', async (req, res) => {
  const search = req.query.search ? req.query.search.trim() : "";

  try {
    // Se não tiver busca → lista tudo
    if (!search) {
      const [results] = await db.query('SELECT * FROM pacientes');
      return res.json(results);
    }

    // Se for número → procura por id
    if (!isNaN(search)) {
      const [results] = await db.query(
        'SELECT * FROM pacientes WHERE idPaciente = ?',
        [search]
      );
      return res.json(results);
    }

    // Se for nome → busca nome parecido
    const [results] = await db.query(
      'SELECT * FROM pacientes WHERE nome LIKE ?',
      [`%${search}%`]
    );

    res.json(results);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});


// ------------------- INSERIR NOVO PACIENTE ------------------- //
router.post('/', async (req, res) => {
  const { nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia } = req.body;
  try {
    const [result] = await db.query(
      'INSERT INTO pacientes (nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia) VALUES (?, ?, ?, ?, ?, ?, ?, ?)',
      [nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia]
    );
    res.status(201).json({
      id: result.insertId,
      nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia
    });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// ------------------- EDITAR PACIENTE ------------------- //
router.put('/:id', async (req, res) => {
  const id = parseInt(req.params.id);
  const { nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia } = req.body;
  try {
    const [result] = await db.query(
      'UPDATE pacientes SET nome=?, dataNascimento=?, telefone=?, email=?, nomeMae=?, idade=?, medicamento=?, patologia=? WHERE idPaciente=?',
      [nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia, id]
    );
    if (result.affectedRows === 0)
      return res.status(404).json({ message: 'Paciente não encontrado' });
    res.json({ id, nome, dataNascimento, telefone, email, nomeMae, idade, medicamento, patologia });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

// ------------------- EXCLUIR PACIENTE ------------------- //
router.delete('/:id', async (req, res) => {
  const id = parseInt(req.params.id);
  try {
    const [result] = await db.query('DELETE FROM pacientes WHERE idPaciente = ?', [id]);
    if (result.affectedRows === 0)
      return res.status(404).json({ message: 'Paciente não encontrado' });
    res.status(204).send();
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;

File: models/java/lab/Paciente.java
package lab;

import java.io.FileWriter;
import java.io.IOException;

public class Paciente {
    private String id;
    private String nome;
    private String dataNascimento;
    private String telefone;
    private String email;
    private String nomeMae;
    private int idade;
    private String medicamento;
    private String patologia;

    public Paciente(String id, String nome, String dataNascimento, String telefone,
                    String email, String nomeMae, int idade, String medicamento, String patologia) {
        this.id = id;
        this.nome = nome;
        this.dataNascimento = dataNascimento;
        this.telefone = telefone;
        this.email = email;
        this.nomeMae = nomeMae;
        this.idade = idade;
        this.medicamento = medicamento;
        this.patologia = patologia;
    }

    public String getId() {
        return id;
    }

    public void salvar() {
        String insert = String.format(
                "INSERT INTO pacientes (idPaciente, nomeCompleto, dataNascimento, idade, telefone, email, nomeMae, nomeMedicamento, nomePatologia) " +
                "VALUES ('%s', '%s', '%s', %d, '%s', '%s', '%s', '%s', '%s');\n",
                id, nome, dataNascimento, idade, telefone, email, nomeMae, medicamento, patologia);

        try (FileWriter writer = new FileWriter("pacientes_inserts.sql", true)) {
            writer.write(insert);
        } catch (IOException e) {
            System.out.println("Erro ao salvar paciente em arquivo: " + e.getMessage());
        }
    }
}
